File: search/TwoCrystalBalls/TwoCrystalBalls.go
package main

import "math"

func TwoCrystalBalls(breaks []bool) int {
	jmpAmount := int(math.Floor(math.Sqrt(float64(len(breaks)))))

	i := jmpAmount

	for ; i < len(breaks); i += jmpAmount {
		if breaks[i] {
			break
		}
	}

	i -= jmpAmount
	for j := 0; j <= jmpAmount && i < len(breaks); j, i = j+1, i+1 {
		if breaks[i] {
			return i
		}
	}
	return -1
}

File: search/TwoCrystalBalls/TwoCrystalBalls.py
import math

def twoCrystalBalls(breaks):
    jmpAmount = math.floor(math.sqrt(len(breaks)))

    i = jmpAmount
    while i < len(breaks):
        if breaks[i]:
            break
        i += jmpAmount

    i -= jmpAmount
    for j in range(jmpAmount + 1):
        if i < len(breaks) and breaks[i]:
            return i
        i += 1

    return -1